core/src/main/scala/mongo4cats/client/ClientSession.scala

```diff
@@ -16,72 +16,63 @@
 
 package mongo4cats.client
 
-import cats.Monad
+import cats.~>
+import cats.arrow.FunctionK
 import cats.effect.Async
-import cats.syntax.alternative._
-import cats.syntax.functor._
+import cats.implicits._
 import com.mongodb.reactivestreams.client.{ClientSession => JClientSession}
 import mongo4cats.helpers._
 
-import scala.util.Try
-
-abstract class ClientSession[F[_]] {
-
-  /** Returns true if there is an active transaction on this session, and false otherwise
-    *
-    * @return
-    *   true if there is an active transaction on this session
-    */
-  def hasActiveTransaction: Boolean
-
-  /** Gets the transaction options. If session has no active transaction, then None is returned
-    *
-    * @return
-    *   the transaction options
-    */
-  def transactionOptions: Option[TransactionOptions]
-
-  /** Start a transaction in the context of this session with the given transaction options. A transaction can not be started if there is
-    * already an active transaction on this session.
-    *
-    * @param options
-    *   the options to apply to the transaction
-    */
-  def startTransaction(options: TransactionOptions): F[Unit]
-  def startTransaction: F[Unit] = startTransaction(TransactionOptions())
-
-  /** Abort a transaction in the context of this session. A transaction can only be aborted if one has first been started.
-    */
+trait ClientSession[F[_]] {
+  def session: JClientSession
```

Comment:
Is it necessary to expose this publicly? Maybe `private[mongo4cats]` if we need to access it internally?

Reply:
:+1: